### PR TITLE
[#1974] fix(hive): Hive table listPartitions reduces HMS RPC calls

### PR DESCRIPTION
**Pull request by cxzl25**
### What changes were proposed in this pull request?
Use HMS's `listPartitions` RPC to replace the two RPC calls of `listPartitionNames` and `getPartitionsByNames`.

### Why are the changes needed?

Fix: #1974

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
exist UT

### CODE REVIEW

File: catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveTableOperations.java [hunk 1]
```diff
@@ -58,30 +60,23 @@ public String[] listPartitionNames() {
...
-    Preconditions.checkArgument(
-        partitionNames.size() == partitions.size(),
-        "oops?! partition names and partitions size are not equal: %s vs %s",
-        partitionNames.size(),
-        partitions.size());
-
-    return IntStream.range(0, partitionNames.size())
-        .mapToObj(i -> fromHivePartition(partitionNames.get(i), partitions.get(i)))
+    List<String> partCols =
+        table.buildPartitionKeys().stream().map(FieldSchema::getName).collect(Collectors.toList());
```
Comment (mchades) [inline catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveTableOperations.java:73]: If the partition keys of the underlying Hive table were changed, `table.buildPartitionKeys()` will get an error result.

Comment (cxzl25) [reply catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveTableOperations.java:73]: Hive table partition keys will not change unless the table is dropped and then created.

File: catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveTableOperations.java [hunk 2]
```diff
@@ -58,30 +60,23 @@ public String[] listPartitionNames() {
...
+
+    return partitions.stream()
+        .map(
+            partition ->
+                fromHivePartition(
+                    FileUtils.makePartName(partCols, partition.getValues()), partition))
```
Comment (mchades) [inline catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveTableOperations.java:79]: Is it necessary to ensure that `partCols` and `partition.getValues()` have corresponding orders?

File: catalogs/catalog-hive/src/test/java/com/datastrato/gravitino/catalog/hive/TestHiveTableOperations.java [hunk 3]
```diff
@@ -41,7 +41,7 @@ public class TestHiveTableOperations extends MiniHiveMetastoreService {
   private static Partition existingPartition;
 
   @BeforeAll
-  private static void setup() {
+  public static void setup() {
```
Comment (mchades) [inline catalogs/catalog-hive/src/test/java/com/datastrato/gravitino/catalog/hive/TestHiveTableOperations.java:44]: Why change the access modifier to `public`?

File: catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveTableOperations.java (hunk 2 above)
Comment (cxzl25) [reply catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveTableOperations.java:79]: Already orderly.

https://github.com/apache/hive/blob/0f8190fefa56513a042566ef87b852e8343610aa/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java#L5243-L5255

File: catalogs/catalog-hive/src/test/java/com/datastrato/gravitino/catalog/hive/TestHiveTableOperations.java (hunk 3 above)
Comment (cxzl25) [reply catalogs/catalog-hive/src/test/java/com/datastrato/gravitino/catalog/hive/TestHiveTableOperations.java:44]: When running related test UT, IDEA has a prompt, so it is modified by the way.

 Here is a non-necessary modification, I can revert.

image

**Review by mchades — APPROVED** [review]
LGTM. Could you please review this when you have time? @jerryshao @FANNG1  Thank you!

**Comment by FANNG1** [comment]
LGTM

**Comment by mchades** [comment]
Thanks @cxzl25  and @FANNG1 , I'm gonna merge this pull request.

**Pull request merged**